cpp/src/centrality/betweenness_centrality_impl.cuh

```diff
@@ -795,18 +796,21 @@ void multisource_backward_pass(
   auto v_first = graph_view.local_vertex_partition_range_first();
 
   // Calculate offsets for each distance level in the consecutive arrays
-  std::vector<size_t> h_distance_offsets(global_max_distance + 1);
+  // Need global_max_distance + 2 elements: one for each distance level (0 to global_max_distance)
+  // plus a sentinel at the end for CUB segmented sort end offsets
+  std::vector<size_t> h_distance_offsets(global_max_distance + 2);
```

Comment:
This appears to be an off-by-one error that results in an illegal memory access in `SortPairs`.